### PR TITLE
v1.4 backports 19-01-23

### PR DESCRIPTION
**Pull request by jrfastab**
PRs:
https://github.com/cilium/cilium/pull/6601
https://github.com/cilium/cilium/pull/6666
https://github.com/cilium/cilium/pull/6621
https://github.com/cilium/cilium/pull/6653
https://github.com/cilium/cilium/pull/6617
https://github.com/cilium/cilium/pull/6661
https://github.com/cilium/cilium/pull/6665
https://github.com/cilium/cilium/pull/6674
https://github.com/cilium/cilium/pull/6677
https://github.com/cilium/cilium/pull/6680
https://github.com/cilium/cilium/pull/6691
https://github.com/cilium/cilium/pull/6694
https://github.com/cilium/cilium/pull/6695
https://github.com/cilium/cilium/pull/6699
https://github.com/cilium/cilium/pull/6703
https://github.com/cilium/cilium/pull/6714
https://github.com/cilium/cilium/pull/6717
https://github.com/cilium/cilium/pull/6719
https://github.com/cilium/cilium/pull/6723

@aanm PR6653 had a series of conflicts in the yaml files. The main difference is we missed marking PR that changes the 'imagePullPolices' in 1.4. So do we want to leave current image Pull policies or update them to align with master.

[EDIT: PR https://github.com/cilium/cilium/pull/6576 was added to resolve merge conflicts]

This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6739)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed [hunk 1]
```diff
@@ -126,6 +126,18 @@ spec:
...
+              name: cilium-config
+              optional: true
+        - name: CILIUM_FLANNEL_UNINSTALL_ON_EXIT
+          valueFrom:
+            configMapKeyRef:
+              key: flannel-uninstall-on-exit
+              name: cilium-config
+              optional: true
         image: docker.io/cilium/cilium:__CILIUM_VERSION__
         imagePullPolicy: IfNotPresent
```
Comment (aanm) [inline examples/kubernetes/templates/v1/cilium-ds.yaml.sed:142]: When you get those conflicts, ignore all `examples/kubernetes/1.*`, fix the conflicts on this file and generate all files again `CILIUM_VERSION=v1.4.0-rc4 make -C examples/kubernetes clean all`. This needs to be done in this commit ("
examples/kubernetes: add options for policy enforcement mode") as the imagePullPolicy should be IfNotPresent

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
@jrfastab it seems something is missing

```
07:14:06 
07:14:06 # github.com/cilium/cilium/test/helpers
07:14:06 helpers/kubectl.go:624:25: cannot use timeout (type time.Duration) as type int64 in field value
07:14:06 
07:14:06 Ginkgo ran 1 suite in 5.099465508s
07:14:06 Test Suite Failed
```

My changes need 92f83a5fb927ca6bc5a0360eea9286faded5c5d0 or tell me and I'll modify commit ("test: add flannel integration in the CI")

**Comment by jrfastab** [comment]
Marking https://github.com/cilium/cilium/pull/6576 as 'needs-backport-v1.4' and pulling in for commit https://github.com/cilium/cilium/commit/92f83a5fb927ca6bc5a0360eea9286faded5c5d0 per @aanm comment above.

**Comment by jrfastab** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -1129,14 +1129,17 @@ func (kub *Kubectl) CiliumEndpointWaitReady() error {
...
+	ctx, cancel := context.WithTimeout(context.Background(), HelperTimeout)
 	defer cancel()
 	err = WithContext(ctx, body, 1*time.Second)
 	if err == nil {
 		return err
 	}
 
 	callback := func() string {
+		ctx, cancel := context.WithTimeout(context.Background(), HelperTimeout)
+		defer cancel()
```
Comment (joestringer) [inline test/helpers/kubectl.go:1141]: I think these lines were actually introduced in another commit, but this matches master so LGTM

(Found the commit later, it actually belongs to `helpers: Limit ep gathering to 5m for all cilium pods`)

**Review by joestringer — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]
Approving for api / docs.

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-docs-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-docs-please

**Comment by jrfastab** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
K8s tests failed: ```02:53:43 Cannot contact jenkins-fixed-1: java.lang.InterruptedException```

Link: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1560

I think this can be merged.

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by tgraf** [comment]
Only failure was known flake that will be fixed in next batch

**Pull request merged**